go/engine/device_keygen.go

```diff
@@ -108,8 +111,14 @@ func (e *DeviceKeygen) SigningKey() libkb.NaclKeyPair {
 	return e.naclSignGen.GetKeyPair()
 }
 
-func (e *DeviceKeygen) EncryptionKey() libkb.NaclKeyPair {
-	return e.naclEncGen.GetKeyPair()
+func (e DeviceKeygen) EncryptionKey() *libkb.NaclDHKeyPair {
+	k := e.naclEncGen.GetKeyPair().(libkb.NaclDHKeyPair)
+	return &k
```

Comment:
I've seen 3 versions of this function
1. `-> NaclKeyPair` on master
2. `-> *libkb.NaclDHKeyPair` some minutes ago in this branch
3. `-> libkb.NaclDHKeyPair` now
I was saying that 2 breaks everything I think by zeroing the key or something. I changed it to 3 which doesn't break any tests. I just don't understand why 2 broke that way nor really why 3 compiles.